diff --git a/arch/arm64/boot/dts/allwinner/sun50i-h618-k2b.dts b/arch/arm64/boot/dts/allwinner/sun50i-h618-k2b.dts
index fc0552e04..c05d6375d 100644
--- a/arch/arm64/boot/dts/allwinner/sun50i-h618-k2b.dts
+++ b/arch/arm64/boot/dts/allwinner/sun50i-h618-k2b.dts
@@ -67,6 +67,15 @@ wifi_pwrseq: pwrseq {
 	};
 };
 
+&emac0 {
+	pinctrl-names = "default";
+	phy-mode = "rgmii";
+	pinctrl-0 = <&ext_rgmii_pins>;
+	phy-handle = <&ext_rgmii_phy>;
+	phy-supply = <&reg_vcc3v3>;
+	status = "okay";
+};
+
 &cpu0 {
 	cpu-supply = <&reg_dcdc2>;
 };
diff --git a/arch/arm64/boot/dts/allwinner/sun50i-h618-k2b.dtsi b/arch/arm64/boot/dts/allwinner/sun50i-h618-k2b.dtsi
index d07e987f6..1c37b9e72 100644
--- a/arch/arm64/boot/dts/allwinner/sun50i-h618-k2b.dtsi
+++ b/arch/arm64/boot/dts/allwinner/sun50i-h618-k2b.dtsi
@@ -62,6 +62,9 @@ &emac0 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&ext_rgmii_pins>;
 	phy-handle = <&ext_rgmii_phy>;
+	allwinner,rx-delay-ps = <2300>;
+	allwinner,tx-delay-ps = <700>;
+	phy-mode = "rgmii";
 	status = "okay";
 };
 
